tests/nc-client/main.c

```diff
@@ -101,6 +103,17 @@ callback_connected(void * cookie, int socket)
 	/* Record socket for future use. */
 	send->socket = socket;
 
+	/* Make socket blocking. */
+	if ((opts = fcntl(send->socket, F_GETFL, 0)) == -1) {
+		warnp("fcntl get");
+		goto err0;
+	}
+	opts = opts & (~O_NONBLOCK);
+	if (fcntl(send->socket, F_SETFL, opts) == -1) {
+		warnp("fcntl set");
+		goto err0;
+	}
+
```

Comment:
I'm confused.  Why do you want blocking networking?

... if you *do* want blocking networking, you would need to stop using all the rest of the libcperciva networking code, since it's designed to deal with non-blocking sockets...